src/components/ColorPicker.tsx

```diff
@@ -42,14 +47,61 @@ const ColorSwatch: FC<{
   const dispatch = useDispatch()
   const colors = useSelector(themeColors)
   const fontSize = useSelector(state => state.fontSize)
-  const selected =
-    (color && cursorStyle?.color === color) || (backgroundColor && cursorStyle?.backgroundColor === backgroundColor)
+  const currentThought = useSelector(state => !!state.cursor && getThoughtById(state, head(state.cursor))) as Thought
   size = size || fontSize * 1.2
 
+  /** A function that adds an alpha channel to a hex color. */
+  const addAlphaToHex = (hex: string) => {
+    if (hex.length === 7) return hex + 'ff'
+    return hex
+  }
+  const colorRegex = /color="(#[0-9a-fA-F]{6})"/g
+  const bgColorRegex = /background-color:\s*(rgb\(\d{1,3},\s?\d{1,3},\s?\d{1,3}\))/g
+  const selected = useMemo(() => {
```

Comment:
Once we get the behavior correct (namely **6. Default background color is not removed**), we should look at how to improve this section. First, it could use some comments to help clarify what it's doing since there's a lot going on. Second, the use of loops and mutations is adding quite a bit of complexity. Maybe we can find a way to simplify and streamline this so that it is easier to reason about.